activity_calendar/models.py

```diff
@@ -332,15 +332,40 @@ def duration(self):
         """ Gets the duration of this activity """
         return self.end_date - self.start_date
 
-    def has_occurrence_at(self, date, is_dst_aware=False):
-        """ Whether this activity has an occurrence that starts at the specified time """
-        if not self.is_recurring:
-            return date == self.start_date
-
-        if not is_dst_aware:
-            date = util.dst_aware_to_dst_ignore(date, self.start_date, reverse=True)
-        occurences = self.get_occurrences_starting_between(date, date)
-        return list(occurences)
+    def get_occurrence_at(self, date, is_dst_aware=False):
+        """
+        Whether this activity has an occurrence that starts at the specified time
+        Note: This does not take shifts in starting moment into account!
+        :param date: Datetime instance of the occurrence
+        :param is_dst_aware: Whether the datetime instance is dst aware (relevant for recurrent activities)
+        :return:
+        """
+        # Check activitymoments on server
+        activitymoments = self.activitymoment_set.filter(recurrence_id=date)
+        if activitymoments.count() == 1:
+            return activitymoments.first()
+        elif activitymoments.count() > 1:
+            raise ActivityMoment.MultipleObjectsReturned()
+
+        # Find recurrent moments that have not yet been created on the server
+        if self.is_recurring:
```

Comment:
(placed here for lack of a better location)
I think we should rewrite `Activity.is_recurring` to also return `True` if there also are activitymoments that don't match the activity's start date. This allows such activities/moments to be handled in the same way as RDATEs, which currently do make an Activity considered to be "recurring".

Reply:
The term is_recurring is currently used as signifier for recurrce patterns only. Rewriting this function might introduce new complexities that I don't want to deal with at the moment. I agree with the remark though :)